src/explore-education-statistics-common/src/utils/url/applyRedirectRules.ts

```diff
@@ -0,0 +1,93 @@
+import redirectService, {
+  RedirectType,
+  redirectPathStarts,
+} from '@common/services/redirectService';
+
+const prodPublicUrl = 'https://explore-education-statistics.service.gov.uk/';
+
+export default async function applyRedirectRules(url: string): Promise<string> {
+  let newUrl: string = url;
+
+  newUrl = await noTrailingSlashUnlessOnlyHost(newUrl);
```

Comment:
If we don't modify as suggested in comment below, then we should remove any unnecessary `awaits`. This is only necessary on the function that calls `redirectService.list`.